Build failure: rclone 1.63.0 on aarch64-darwin

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `rclone` 1.63.0 on `aarch64-darwin`

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/5xqqv8l0b4iq0cl22q7nhvbylcd9z0rh-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage ./.
# github.com/shoenig/go-m1cpu
vendor/github.com/shoenig/go-m1cpu/cpu.go:65:33: error: use of undeclared identifier 'kIOMainPortDefault'; did you mean 'kIOMasterPortDefault'?
   IOServiceGetMatchingServices(kIOMainPortDefault, matching, &iter);
                                ^~~~~~~~~~~~~~~~~~
                                kIOMasterPortDefault
/nix/store/kpkqip48d0k2capsw9xq4ilr5ph6j5yp-apple-framework-IOKit-11.0.0/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:103:19: note: 'kIOMasterPortDef>
const mach_port_t kIOMasterPortDefault;
                  ^
1 error generated.
```

### Additional context

- rclone 1.63.0 is available in nixos-unstable

### Notify maintainers

@danielfullmer @marsam @SuperSandro2000

### Metadata

Using flakes, tracking nixos-unstable

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.5.0, macOS 13.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - channels(cdom): `"home-manager"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/d8myz5q96bls7a8yvrq94sdmn41xhfqw-2p1jsxsxv0d0gkybanb5acbpi7sr6rjw-source`
```

I opened an issue under `nix`, which was incorrect, but I have the same issue. This is what my ticket said. I think the issue maybe with more than just rclone. 

Describe the bug
There appears to be an issue with packages that use the later versions of github.com/shoenig/go-m1cpu. I don't think the issue is with github.com/shoenig/go-m1cpu directly. See https://github.com/shoenig/go-m1cpu/issues/5

So far the error can be seen in both nomad and rclone. See the example error below.(Ran nix-build -E 'with import <nixpkgs> { }; callPackage ./default.nix { }' from inside nomad folder on master branch.

 nix-build -E 'with import <nixpkgs> { }; callPackage ./default.nix { }'
this derivation will be built:
  /nix/store/psdaf1j3fk5z7z7pqxi53rzpkdd3aj0y-nomad-1.5.6.drv
these 2 paths will be fetched (57.67 MiB download, 158.16 MiB unpacked):
  /nix/store/w617m8r5pcp96g50k4lwzai4ddjpj5b3-nomad-1.2.16
  /nix/store/rv50wv3wbfafbdvllzkj7bd51s3i39fv-nomad-1.3.9
copying path '/nix/store/rv50wv3wbfafbdvllzkj7bd51s3i39fv-nomad-1.3.9' from 'https://cache.nixos.org'...
copying path '/nix/store/w617m8r5pcp96g50k4lwzai4ddjpj5b3-nomad-1.2.16' from 'https://cache.nixos.org'...
building '/nix/store/psdaf1j3fk5z7z7pqxi53rzpkdd3aj0y-nomad-1.5.6.drv'...
unpacking sources
unpacking source archive /nix/store/p0c06jzmy24wkhg5q2ydpipdkck9gyvp-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
building
Building subPackage ./.
# github.com/shoenig/go-m1cpu
vendor/github.com/shoenig/go-m1cpu/cpu.go:65:33: error: use of undeclared identifier 'kIOMainPortDefault'; did you mean 'kIOMasterPortDefault'?
   IOServiceGetMatchingServices(kIOMainPortDefault, matching, &iter);
                                ^~~~~~~~~~~~~~~~~~
                                kIOMasterPortDefault
/nix/store/fqlxrzgpr6hrkbrb060g6njfn5yr7b1y-apple-framework-IOKit-11.0.0/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:103:19: note: 'kIOMasterPortDefault' declared here
const mach_port_t kIOMasterPortDefault;
                  ^
1 error generated.
error: builder for '/nix/store/psdaf1j3fk5z7z7pqxi53rzpkdd3aj0y-nomad-1.5.6.drv' failed with exit code 1;
       last 10 log lines:
       > Building subPackage ./.
       > # github.com/shoenig/go-m1cpu
       > vendor/github.com/shoenig/go-m1cpu/cpu.go:65:33: error: use of undeclared identifier 'kIOMainPortDefault'; did you mean 'kIOMasterPortDefault'?
       >    IOServiceGetMatchingServices(kIOMainPortDefault, matching, &iter);
       >                                 ^~~~~~~~~~~~~~~~~~
       >                                 kIOMasterPortDefault
       > /nix/store/fqlxrzgpr6hrkbrb060g6njfn5yr7b1y-apple-framework-IOKit-11.0.0/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:103:19: note: 'kIOMasterPortDefault' declared here
       > const mach_port_t kIOMasterPortDefault;
       >                   ^
       > 1 error generated.
       For full logs, run 'nix log /nix/store/psdaf1j3fk5z7z7pqxi53rzpkdd3aj0y-nomad-1.5.6.drv'.
So far I get the same issue on both Rclone and Nomad packages. This seems like a relatively new issue as both Rclone and Nomad have recently updated the github.com/shoenig/go-m1cpu package

I believe this issue is related to nix and NOT the go-m1cpu as i'm able to checkout both rclone and nomad source repositories and run go build successfully.

Steps To Reproduce

Try building latest version of nomad or rclone on an m2 mac using the latest nix derivation located at pkgs/applications/networking/sync/rclone/ and pkgs/applications/networking/cluster/nomad/
Example command nix-build -E 'with import <nixpkgs> { }; callPackage ./default.nix { }'
Expected behavior

Packages build correctly

nix-env --version output
nix-env (Nix) 2.15.1

Additional context
OSX 13.4.1, M2

--- a/pkgs/applications/networking/sync/rclone/default.nix
+++ b/pkgs/applications/networking/sync/rclone/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, buildGoModule, fetchFromGitHub, buildPackages, installShellFiles
+{ lib, stdenv, buildGoModule, fetchFromGitHub, buildPackages, installShellFiles, fetchpatch
 , makeWrapper
 , enableCmount ? true, fuse, macfuse-stubs
 }:
@@ -14,7 +14,16 @@ buildGoModule rec {
     hash = "sha256-ojP1Uf9iP6kOlzW8qsUx1SnMRxFZLsgkjFD4LVH0oTI=";
   };
 
-  vendorSha256 = "sha256-0YenfRa5udTrajPLI1ZMV+NYDHKO++M0KvIvr4gYLLc=";
+  patches = [
+    # Fix build on aarch64-darwin. Remove with the next release.
+    # https://github.com/rclone/rclone/pull/7099
+    (fetchpatch {
+      url = "https://github.com/rclone/rclone/commit/fb5125ecee4ae1061ff933bb3b9b19243e022241.patch";
+      hash = "sha256-3SzU9iiQM8zeL7VQhmq0G6e0km8WBRz4BSplRLE1vpM=";
+    })
+  ];
+
+  vendorSha256 = "sha256-AXgyyI6ZbTepC/TGkHQvHiwpQOjzwG5ung71nKE5d1Y=";
 
   subPackages = [ "." ];
 

